griptape/tools/griptape_cloud_structure_run_client/tool.py

```diff
@@ -0,0 +1,99 @@
+from __future__ import annotations
+import time
+from typing import Any, Optional
+from urllib.parse import urljoin
+from schema import Schema, Literal
+from attr import define, field
+from griptape.tools.base_griptape_cloud_client import BaseGriptapeCloudClient
+from griptape.utils.decorators import activity
+from griptape.artifacts import InfoArtifact, TextArtifact, ErrorArtifact
+
+
+@define
+class GriptapeCloudStructureRunClient(BaseGriptapeCloudClient):
+    """
+    Attributes:
+        description: LLM-friendly structure description.
+        structure_id: ID of the Griptape Cloud Structure.
+    """
+
+    _description: Optional[str] = field(default=None, kw_only=True)
+    structure_id: str = field(kw_only=True)
+    structure_run_wait_time_interval: int = field(default=2, kw_only=True)
+    structure_run_max_wait_time_attempts: int = field(default=20, kw_only=True)
+
+    @activity(
+        config={
+            "description": "Can be used to execute a Run of a Structure with the following description: {{ _self.description }}",
+            "schema": Schema(
+                {Literal("args", description="A list of string arguments to submit to the Structure Run"): list}
+            ),
+        }
+    )
+    def execute_structure_run(self, params: dict) -> InfoArtifact | TextArtifact | ErrorArtifact:
+        from requests import post, exceptions, HTTPError, Response
+
+        args: list[str] = params["values"]["args"]
+        url = urljoin(self.base_url.strip("/"), f"/api/structures/{self.structure_id}/runs")
+
+        try:
+            response: Response = post(url, json={"args": args}, headers=self.headers)
+            response.raise_for_status()
+            response_json = response.json()
+            return self._get_structure_run_result(response_json["structure_run_id"])
+
+        except (exceptions.RequestException, HTTPError) as err:
+            return ErrorArtifact(str(err))
+
+    def _get_structure_run_result(self, structure_run_id: str) -> InfoArtifact | TextArtifact | ErrorArtifact:
+        url = urljoin(self.base_url.strip("/"), f"/api/structure-runs/{structure_run_id}")
+
+        result = self._get_structure_run_result_attempt(url)
+        status = result.get("status")
```

Comment:
This has the potential to be `None`. We should use a regular dictionary lookup or handle the `None` case if we ever expect it to be so.